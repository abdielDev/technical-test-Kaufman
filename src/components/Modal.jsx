import React from 'react';
import ReactDOM from 'react-dom';
import '../css/components/Modal.css';

const Modal = ({ isOpen, onClose }) => {

  if(!isOpen) {
    return null
  }
  return ReactDOM.createPortal(
    <div className="Modal">
      <div className="Modal--container">
        <button onClick={onClose} className="Modal--close-button">
            X
        </button>
        <h1>Are you sure about this purchase?</h1>
        <div className="Modal--Buttons">
          <button onClick={onClose} className="Modal--Button">Yes</button>
          <button onClick={onClose} className="Modal--Button">No</button>
        </div>
      </div>
    </div>,
    document.getElementById('modal')
  );
};

export default Modal;