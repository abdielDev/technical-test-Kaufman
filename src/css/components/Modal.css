.Modal {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  flex-direction: column;
  align-items: center;
  color: var(--color-black-primary);
}

.Modal--container {
  position: relative;
  top: 2rem;
  background-color: #ffffff;
  width: 500px;
}

.Modal--container {
  padding: 1em;
}

.Modal--content {
  padding: 1em;
  text-align: center;
}

.Modal--content h2 {
  margin-bottom: 10px;
}

.Modal--content p {
  text-align: justify;
  margin-bottom: 30px;
}

.Modal--content a {
  padding: 5px;
  color: #ffffff;
  background-color: blueviolet;
  text-decoration: none;
}

.Modal--close-button {
  position: absolute;
  top: 0;
  right: 0;
  border: 0;
  background-color: #F5F5F5;
  padding: 0.5rem 1rem;
  cursor: pointer;
  outline-style: none;
}

.Modal--Buttons {
  text-align: center;
}

.Modal--Buttons button:first-child {
  margin-right: 20px;
}

.Modal--Button {
  padding: 10px;
  background-color: var(--color-button-primary);
  border: none;
  cursor: pointer;
  font-weight: bold;
  color: var(--color-white-primary);
}