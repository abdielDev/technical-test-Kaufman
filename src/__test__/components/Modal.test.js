import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Modal from '../../components/Modal';
import { expect, jest } from '@jest/globals';
import ReactDOM from 'react-dom';

describe('<Modal />', () => {
  let component;

  // add a div with #modal-root id to the global body
  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'modal');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);
  test('Component Modal render', () => {
    component = mount(<Modal isOpen />);
    expect(component.length).toEqual(1);
  });

  test('It not received the mission details props', () => {
    component = mount(<Modal isOpen />);
    expect(component.find('h1').text()).toBe('Are you sure about this purchase?');
  });

  test('Should trigger onClose when clicked', () => {
    const onClose = jest.fn();
    component = mount(<Modal isOpen onClose={onClose} />);

    component.find('.Modal--close-button').simulate('click');
    expect(onClose.mock.calls).toHaveLength(1)
  })

  afterEach(() => {
    component.unmount();
  });
});

describe('Modal Snapshot', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear()
  });
  test('Check UI of component Modal', () => {
    const modal = create(<Modal isOpen />);
    expect(modal.toJSON()).toMatchSnapshot();
  });
});