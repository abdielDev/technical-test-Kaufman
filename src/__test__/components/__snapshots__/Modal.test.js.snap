// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshot Check UI of component Modal 1`] = `
<div
  className="Modal"
>
  <div
    className="Modal--container"
  >
    <button
      className="Modal--close-button"
    >
      X
    </button>
    <h1>
      Are you sure about this purchase?
    </h1>
    <div
      className="Modal--Buttons"
    >
      <button
        className="Modal--Button"
      >
        Yes
      </button>
      <button
        className="Modal--Button"
      >
        No
      </button>
    </div>
  </div>
</div>
`;
